libs/ssl/threading_alt.h

```diff
@@ -1,3 +1,4 @@
+#define WIN32_LEAN_AND_MEAN
```

Comment:
Huh, I thought this define was to exclude some things. How does this "fix" anything?

Reply:
Hmm, I mean this is a good define to make regardless, I just find it strange that it's strictly necessary for compilation. But then again it probably won't matter. Thanks!

Reply:
It's just about the order of includes. Including `windows.h` in `threading_alt.h` like we do results in this ordering, which breaks compilation:

```c
#include <windows.h> // threading_alt.h
// ...
#include <winsock2.h> // mbedtls sources
```

Since at the time when `windows.h` is included we haven't included `winsock2.h` yet, `windows.h` includes `winsock.h` instead which contains conflicting definitions. Then when `winsock2.h` is included we get redefinition errors.

If `WIN32_LEAN_AND_MEAN` is defined, then `winsock.h` is excluded from `windows.h` so we don't end up with these redefinition errors when including `winsock2.h`.

More details here: https://stackoverflow.com/questions/21399650/cannot-include-both-files-winsock2-windows-h